kafka/consumer.go

```diff
@@ -420,18 +438,27 @@ func NewConsumer(conf *ConfigMap) (*Consumer, error) {
 
 	c.handle.c = c
 	c.handle.setup()
+	c.name = c.handle.String()
+	c.events = make(chan Event, eventsChanSize)
+	c.logEvents = make(chan Event, eventsChanSize)
+	c.readerTermChan = make(chan bool)
 	c.handle.rkq = C.rd_kafka_queue_get_consumer(c.handle.rk)
 	if c.handle.rkq == nil {
 		// no cgrp (no group.id configured), revert to main queue.
 		c.handle.rkq = C.rd_kafka_queue_get_main(c.handle.rk)
 	}
 
-	if c.eventsChanEnable {
-		c.events = make(chan Event, eventsChanSize)
-		c.readerTermChan = make(chan bool)
+	if logForward {
```

Comment:
This should be moved to the Handle implementation so it can be reused for the Producer